consensus/src/processes/pruning_proof/mod.rs

```diff
@@ -743,12 +744,26 @@ impl PruningProofManager {
         level_depth << current_dag_level.saturating_sub(level)
     }
 
+    /// selected parent at level = the parent of the header at the level
+    /// with the highest blue_work (using score as work in this case)
     fn find_selected_parent_header_at_level(
         &self,
         header: &Header,
         level: BlockLevel,
+        relations_service: MTRelationsService<DbRelationsStore>,
     ) -> PruningProofManagerInternalResult<Arc<Header>> {
-        let parents = self.parents_manager.parents_at_level(header, level);
+        // Logic of apply_proof only inserts parent entries for a header from the proof
+        // into the relations store for a level if there was GD data in the old stores for that
+        // header. To mimic that logic here, we need to filter out parents that are NOT in the relations_service
```

Comment:
Don't we add all blocks to `relations` in `apply_proof`?

Reply:
We apply all the blocks that are in the proof. Turns out, the logic being applied in the filtering in `apply_proof`. The filtering logic in `apply_proof` is this (in master):

```
            for header in headers.iter() {
                let parents = Arc::new(
                    self.parents_manager
                        .parents_at_level(header, level as BlockLevel)
                        .iter()
                        .copied()
                        .filter(|parent| self.ghostdag_stores[level].has(*parent).unwrap())
                        .collect_vec()
                        .push_if_empty(ORIGIN),
                );

                self.relations_stores.write()[level].insert(header.hash, parents.clone()).unwrap();
```

From reading the code around this in `apply_proof`, it looks like the only reason to do the filtering is to ensure that only the "relevant parents" (those that are included in the proof, and are consequently added to the ghostdag_store) of any block are considered.

I've updated the comment in this PR to read:

```
        // Parents manager parents_at_level may return parents that aren't in relations_service, so it's important
        // to filter to include only parents that are in relations_service.
        let parents = self
            .parents_manager
            .parents_at_level(header, level)
            .iter()
            .copied()
            .filter(|parent| relations_service.has(*parent).unwrap())
            .collect_vec()
            .push_if_empty(ORIGIN);
```